rllib/examples/policy/episode_env_aware_policy.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+
+from ray.rllib.agents.callbacks import DefaultCallbacks
+from ray.rllib.examples.policy.random_policy import RandomPolicy
+from ray.rllib.policy.policy import Policy
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.policy.view_requirement import ViewRequirement
+from ray.rllib.utils.annotations import override
+
+
+class EpisodeEnvAwareCallback(DefaultCallbacks):
+    def on_episode_start(self, worker, base_env, policies, episode, **kwargs):
+        policies["pol0"].episode_id = episode.episode_id
+        policies["pol0"].env_id = worker.env_context.vector_index
+
+
+class EpisodeEnvAwarePolicy(RandomPolicy):
+    """A Policy that always knows the current EpisodeID and EnvID and
+    returns these in its actions."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.episode_id = None
+        self.env_id = None
+        class _fake_model: pass
+        self.model = _fake_model()
+        self.model.inference_view_requirements = {
+            SampleBatch.OBS: ViewRequirement(),
+            SampleBatch.PREV_ACTIONS: ViewRequirement(
+                SampleBatch.ACTIONS, space=self.action_space, shift=-1),
+            SampleBatch.PREV_REWARDS: ViewRequirement(
+                SampleBatch.REWARDS, shift=-1),
```

Comment:
I was kind of expecting EPS_ID and ENV_ID to be in this view based on the class comment, and not prev action/reward is this an error?

Reply:
Good point. Makes the Callback superfluous as well. Changed.